Model/Sync/Reset/Catalog.php

```diff
@@ -0,0 +1,143 @@
+use Yotpo\Core\Model\Config as CoreConfig;
+
+class Catalog extends Main
+{
+    const PRODUCT_SYNC_TABLE = 'yotpo_product_sync';
+    const CATEGORY_SYNC_TABLE = 'yotpo_category_sync';
+    const CRONJOB_CODES = ['yotpo_cron_core_category_sync','yotpo_cron_core_products_sync'];
+    const YOTPO_ENTITY_NAME = 'catalog';
+
+    /**
+     * @var SyncDataMain
+     */
+    protected $syncDataMain;
+
+    /**
+     * @param ResourceConnection $resourceConnection
+     * @param SyncDataMain $syncDataMain
+     */
+    public function __construct(
+        ResourceConnection $resourceConnection,
+        Config $config,
+        TypeListInterface $cacheTypeList,
+        SyncDataMain $syncDataMain
+    ) {
+        parent::__construct(
+            $resourceConnection,
+            $config,
+            $cacheTypeList
+        );
+        $this->syncDataMain = $syncDataMain;
+    }
+
+    /**
+     * @return string
+     */
+    public function getYotpoEntityName()
+    {
+        return self::YOTPO_ENTITY_NAME;
+    }
+
+    /**
+     * @param int $storeId
+     * @param boolean $skipSyncTables
+     * @return void
+     * @throws \Zend_Db_Statement_Exception
+     */
+    public function resetSync($storeId, $skipSyncTables = false)
+    {
+        parent::resetSync($storeId);
+        $this->resetCatalogSyncAttributes($storeId);
+        $this->setResetInProgressConfig($storeId, '0');
+    }
+
+    /**
+     * @return array <string>
+     */
+    protected function getTableResourceNames()
+    {
+        return [self::PRODUCT_SYNC_TABLE, self::CATEGORY_SYNC_TABLE];
+    }
+
+    /**
+     * @return array <string>
+     */
+    protected function getCronJobCodes()
+    {
+        return self::CRONJOB_CODES;
+    }
+
+    /**
+     * @param int $storeId
+     * @return void
+     */
+    private function resetCatalogSyncAttributes($storeId)
+    {
+        $dataSet = [
+            [
+                'table_name' => 'catalog_category_entity_int',
+                'attribute_code' => CoreConfig::CATEGORY_SYNC_ATTR_CODE
+            ],
+            [
+                'table_name' => 'catalog_product_entity_int',
+                'attribute_code' => CoreConfig::CATALOG_SYNC_ATTR_CODE
+            ],
+        ];
+        foreach ($dataSet as $data) {
+            $connection =   $this->resourceConnection->getConnection();
+            $attributeId = $this->syncDataMain->getAttributeId($data['attribute_code']);
+            $totalCount = $this->getCountOfEntities($data['table_name'], $attributeId, $storeId);
+            $tableName = $this->resourceConnection->getTableName($data['table_name']);
+            while ($totalCount > 0) {
+                if ($totalCount > self::UPDATE_LIMIT) {
+                    $limit = self::UPDATE_LIMIT;
+                    $totalCount -= self::UPDATE_LIMIT;
+                } else {
+                    $limit = $totalCount;
+                    $totalCount = 0;
+                }
```

Comment:
This is not actually necessary if you're decreasing the limit from `totalCount` at the end of the while loop.

Reply:
This is implemented. Please check the updated PR